core/src/style/textStyleBuilder.cpp

```diff
@@ -189,12 +189,13 @@ bool TextStyleBuilder::addFeature(const Feature& _feat, const DrawRule& _rule) {
 
     } else if (_feat.geometryType == GeometryType::polygons) {
 
-        for (auto& polygon : _feat.polygons) {
-            for (auto& line : polygon) {
-                for (auto& point : line) {
-                    addLabel(params, Label::Type::point, { point }, _rule);
-                }
+        glm::vec3 c;
+        const auto& polygons = _feat.polygons;
+        for (const auto& polygon : polygons) {
+            if (!polygon.empty()) {
+                c = centroid(polygon.front().begin(), polygon.front().end());
             }
+            addLabel(params, Label::Type::point, { c }, _rule);
```

Comment:
Why declare `c` out of the loop? We risk to use it initialized to `(0,0,0)`.

We should probably put this statement in the condition.